central/complianceoperator/v2/report/manager/watcher/scanconfigwatcher.go

```diff
@@ -127,21 +147,29 @@ func (w *scanConfigWatcherImpl) run(timer Timer) {
 			log.Infof("Stopping scan config watcher")
 			return
 		case <-timer.C():
-			log.Warnf("Timeout waiting for the ScanConfiguration %s's scans to finish", w.scanConfigResults.ScanConfig.GetScanConfigName())
-			concurrency.WithLock(&w.snapshotsLock, func() {
+			concurrency.WithLock(&w.resultsLock, func() {
+				log.Warnf("Timeout waiting for the ScanConfiguration %s's scans to finish", w.scanConfigResults.ScanConfig.GetScanConfigName())
 				w.scanConfigResults.Error = ScanConfigTimeoutError
 				w.readyQueue.Push(w.scanConfigResults)
 			})
 			return
 		case result := <-w.scanC:
 			if err := w.handleScanResults(result); err != nil {
 				log.Errorf("Unable to handle scan results %s: %v", result.Scan.GetScanName(), err)
+				concurrency.WithLock(&w.resultsLock, func() {
+					w.scanConfigResults.Error = err
+					w.readyQueue.Push(w.scanConfigResults)
+				})
+				return
 			}
 		}
-		if w.totalResults != 0 && w.totalResults == len(w.scanConfigResults.ScanResults) {
-			concurrency.WithLock(&w.snapshotsLock, func() {
+		if concurrency.WithLock1[bool](&w.resultsLock, func() bool {
+			if w.totalResults != 0 && w.totalResults == len(w.scanConfigResults.ScanResults) {
 				w.readyQueue.Push(w.scanConfigResults)
```

Comment:
I think, we may need to protect all access to `w.totalResults` with a mutex. I understand that `resultsLock` protects  `w.scanConfigResults.ScanResults`, but if there is a chance that `handleScanResults` runs concurrently, then we need to do that for `totalResults` too.

Reply:
I think this is not needed - results from confusion of `ComplianceOperatorCheckResultV2` with `ScanWatcherResults`.